### DISCUSSION TITLE
Motive 2.0 with VRPN streaming (Unknown motion capture type!

### DISCUSSION THREAD

**Discussion by elim17** [USC-ACTLab/crazyswarm#299] [Q&A]
Hello,

I was referencing the issue below as our lab is using vrpn with motive Optitrack.[https://github.com/USC-ACTLab/crazyswarm/issues/121](url). When I run hover_swarm.launch the server dies because of a runtime_error "Unknown motion capture type". I know it is the motion capture type parameter in the launch file, because when I change the parameter to Optitrack,  the server launches and stays running. However, when I hit the start button on the controller it still doesn't fly. Any ideas?

**Comment by whoenig** [comment]
I think VRPN is not compiled by default anymore, because of issues with Ubuntu 20.04/ROS melodic. However, there is no reason to use VRPN for an Optitrack system - the integrated optitrack support should be far superior to the VRPN mode.

To check your mocap, you can try launching `mocap_helper.launch` instead of `hover_swarm.launch`. If your drone isn't doing anything, it is likely a firmware issue, though - did you upgrade your Crazyradio as well as NRF firmwares?

**Comment by wangmenghetrue** [comment]
Hi! Has your problem been solved? The motion capture system I use also uses vrpn. Then, I ran hover_swarm.launch and had the same problem as you. I ran mocap_helper.launch and showed the following results.`what(): Unknown motion capture type !`
My configuration file is
` # tracking`
`motion_capture_type: "vrpn" # one of none,vicon,optitrack,qualisys,vrpn`
`object_tracking_type: "motionCapture" # one of motionCapture,libobjecttracker`
`send_position_only: False # set to False to send position+orientation; set to True to send position only`
`#vicon_host_name: "vicon"`
`# optitrack_host_name: "optitrack"`
`# qualisys_host_name: "10.0.5.219"`
`# qualisys_base_port: 22222`
`vrpn_host_name: "192.168.2.137"`
`save_point_clouds: ~/pointCloud.ot`
`print_latency: False`
I really hope to get your help! thank you very much!

**Reply by whoenig** [reply]
See https://github.com/USC-ACTLab/crazyswarm/issues/505.

**Comment by elim17** [comment]
Hi,

Unfortunately we decided to use the Motive Optitrack's default streaming
which uses NatNet. It seems that the VRPN streaming has been deprecated.
What mocap software are you using?
